staging/src/k8s.io/apiserver/pkg/registry/rest/update.go

```diff
@@ -99,6 +101,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx genericapirequest.Context, ob
 	}
 	objectMeta.SetGeneration(oldMeta.GetGeneration())
 
+	// Ensure Initializers are not set unless the feature is enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(features.Initializers) {
+		oldMeta.SetInitializers(nil)
```

Comment:
Why setting the oldMeta?

Reply:
if the feature is disabled, the server should behave as if it doesn't know about the field at all. ideally that would be done in decoding based on field gates, but until then, we mimic that by dropping it from all objects here.